wp-dependency-installer.php

```diff
@@ -50,18 +65,30 @@ class WP_Dependency_Installer {
 		 *
 		 * @var array $notices
 		 */
-		protected $notices = [];
+		protected $notices;
 
 		/**
-		 * Singleton.
+		 * Multiton.
+		 *
+		 * @param string $plugin_path (optional) Path to plugin or theme calling the framework.
 		 */
-		public static function instance() {
-			static $instance = null;
-			if ( null === $instance ) {
-				$instance = new self();
+		public static function instance( $plugin_path = '' ) {
+			if ( ! array_key_exists( $plugin_path, self::$instances ) ) {
+				self::$instances[ $plugin_path ] = new self( $plugin_path );
 			}
+			return self::$instances[ $plugin_path ];
+		}
 
-			return $instance;
+		/**
+		 * Private constructor.
+		 *
+		 * @param string $plugin_path Path to plugin or theme calling the framework.
+		 */
+		protected function __construct( $plugin_path ) {
+			$this->plugin_path = $plugin_path;
+			$this->source      = basename( $plugin_path );
+			$this->config      = [];
```

Comment:
Do you think a static variable could be better? ( currently each instance has its own config array )

Reply:
What we need to consider is multiple sources using the framework at the same time.

Is there any need to save anything other that a basic instance?

Reply:
You mean `static::$plugin_path` vs `$this->plugin_path` ?

Reply:
> Is there any need to save anything other that a basic instance?

I was thinking about it too, most likely the singleton solution is still the most suitable.

In order to avoid too much collateral issues, we can also only use the following version:

```php
WP_Dependency_Installer::instance()->register( $config, __DIR__ );
WP_Dependency_Installer::instance()->run( __DIR__ );
```

Reply:
Hmm, I actually liked being able to pass the $plugin_path to `instance()`

Reply:
Maybe save that _feature_ for later. 🙃

Reply:
> Hmm, I actually liked being able to pass the $plugin_path to `instance()`

Me too, but I think the code could easily have become too complicated (thus losing the singleton's only advantage, simple access to "status" variables <-- i.e. global config )

> Maybe save that feature for later. 🙃

Here it is: https://github.com/Raruto/wp-dependency-installer/blob/6ee379b432b50afce4b17135ce2a637ee5e9455c/wp-dependency-installer.php

( later I will add within the main description of this pull )